app/src/test/java/com/example/habitsmasher/test/ListenersTest.java

```diff
@@ -0,0 +1,135 @@
+package com.example.habitsmasher.test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.fragment.app.FragmentManager;
+
+import com.example.habitsmasher.DaysTracker;
+import com.example.habitsmasher.HabitList;
+import com.example.habitsmasher.User;
+import com.example.habitsmasher.listeners.ClickListenerForCancel;
+import com.example.habitsmasher.listeners.ClickListenerForDatePicker;
+import com.example.habitsmasher.listeners.ClickListenerForDaysOfTheWeek;
+import com.example.habitsmasher.listeners.FailureListener;
+import com.example.habitsmasher.listeners.FailureListenerWithToast;
+import com.example.habitsmasher.listeners.SuccessListener;
+import com.example.habitsmasher.listeners.SuccessListenerWithToast;
+import com.example.habitsmasher.listeners.SwipeListener;
+import com.example.habitsmasher.ui.dashboard.HabitItemAdapter;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * This class hosts the unit tests for the listeners
+ *
+ * The reference for testing listeners is the following url:
+ * URL: https://stackoverflow.com/questions/18913827/how-to-test-listener-interface-is-called-within-android-unit-tests/19050763
+ */
+public class ListenersTest {
```

Comment:
I would prefer if the tests corresponding to **each class** were split up into their own files. Even if it's just one/two tests per file, it just really helps when you need to look for the tests for **that specific class**. This shouldn't be too bad, and makes the test class a lot easier to follow and modify in the future.
- [ ] move unit tests to their own test class for each listener